src/permission-set.js

```diff
@@ -564,6 +628,43 @@ class PermissionSet {
   }
 
   /**
+   * Returns a list of webIds of groups to which this agent belongs.
+   * Note: Only checks loaded groups (assumes a previous `loadGroups()` call).
```

Comment:
I think the loading of groups should be transparent to the caller. It should not be the responsibility of the client to handle/understand the internal implementation of the library.